Remotion/Data/DomainObjects/Infrastructure/Serialization/FlattenedDeserializationInfo.cs

```diff
@@ -73,7 +73,14 @@ public T GetValue<T> ()
     [return: MaybeNull]
     private T GetValue<T> (bool allowNull)
     {
-      if (typeof(IFlattenedSerializable).IsAssignableFrom(typeof(T)))
+      if (typeof(T) == typeof(Type))
+      {
+        var typeName = GetValue<string>(allowNull);
+        return typeName != null
+            ? (T)(object)Type.GetType(typeName, throwOnError: true, ignoreCase: false)!
```

Comment:
We should refactor to extract the typename conversion into a helper method to remove the code duplication.

Unrelated: is .net smart enough to allow the use of `(Type)` instead of `(T)(object)`?

Reply:
Sadly, .NET is not smart enough so we need to do the cast to `object` first, thus we have a bit of code clutter.